libbeat/autodiscover/providers/kubernetes/pod.go

```diff
@@ -75,43 +75,34 @@ func NewPodEventer(uuid uuid.UUID, cfg *common.Config, client k8s.Interface, pub
 		return nil, fmt.Errorf("couldn't create watcher for %T due to error %+v", &kubernetes.Pod{}, err)
 	}
 
-	var nodeMeta, namespaceMeta metadata.MetaGen
-	var nodeWatcher, namespaceWatcher kubernetes.Watcher
+	options := kubernetes.WatchOptions{
+		SyncTimeout: config.SyncPeriod,
+		Node:        config.Node,
+	}
+	if config.Namespace != "" {
+		options.Namespace = config.Namespace
+	}
 	metaConf := config.AddResourceMetadata
-	if metaConf != nil {
-		if metaConf.Node != nil && metaConf.Node.Enabled() {
```

Comment:
The options to enable node and namespace metadata will be ignored now, we should mention it somewhere and/or remove them from the docs.

Reply:
Removed completely.